src/test/java/com/atlassian/jira/cloud/jenkins/listeners/ConfigurationSaveableListenerTest.java

```diff
@@ -0,0 +1,107 @@
+
+        listener = new ConfigurationSaveableListener();
+        listener.setSecretRetriever(secretRetriever);
+        listener.setConnectionDataApi(pluginConfigApi);
+        listener.setXmlUtils(xmlUtils);
+    }
+
+    @Test
+    public void testOnChange() {
+        // Arrange
+        Saveable saveable = Mockito.mock(JiraCloudPluginConfig.class);
+        XmlFile file = Mockito.mock(XmlFile.class);
+
+        ConfigurationSaveableListener configurationSaveableListener = new ConfigurationSaveableListener();
+        ConfigurationSaveableListener spyListener = Mockito.spy(configurationSaveableListener);
+
+        Mockito.doNothing().when(spyListener).sendPluginConfigData(file);
+
+        // Act
+        spyListener.onChange(saveable, file);
+
+        // Assert
+        Mockito.verify(spyListener).sendPluginConfigData(file);
+    }
+
+    @Test
+    public void testSendPluginConfigData_SuccessfullySendsDataToPluginApi() throws Exception {
+        // Arrange
+        XmlFile file = mock(XmlFile.class);
+        Element rootElement = mock(Element.class);
+        NodeList siteElements = mock(NodeList.class);
+        Element siteElement = mock(Element.class);
+
+        when(xmlUtils.parseXmlFile(file)).thenReturn(rootElement);
+        when(xmlUtils.extractXmlValue(rootElement, "autoBuildsRegex")).thenReturn("regex");
+        when(xmlUtils.extractBooleanValue(rootElement, "autoBuildsEnabled")).thenReturn(true);
+        when(xmlUtils.extractBooleanValue(rootElement, "autoDeploymentsEnabled")).thenReturn(true);
+        when(xmlUtils.extractXmlValue(rootElement, "autoDeploymentsRegex")).thenReturn("regex");
+        when(rootElement.getElementsByTagName("atl-jsw-site-configuration")).thenReturn(siteElements);
+        when(siteElements.getLength()).thenReturn(1);
+        when(siteElements.item(0)).thenReturn(siteElement);
+        when(xmlUtils.extractXmlValue(siteElement, "webhookUrl")).thenReturn("url");
+        when(xmlUtils.extractXmlValue(siteElement, "credentialsId")).thenReturn("id");
+        when(secretRetriever.getSecretFor(eq("id"))).thenReturn(Optional.of("secret"));
+        when(xmlUtils.parseXmlFile(file)).thenReturn(rootElement);
+        when(xmlUtils.extractXmlValue(rootElement, "autoBuildsRegex")).thenReturn("regex");
+        when(xmlUtils.extractBooleanValue(rootElement, "autoBuildsEnabled")).thenReturn(true);
+        when(xmlUtils.extractBooleanValue(rootElement, "autoDeploymentsEnabled")).thenReturn(true);
+        when(xmlUtils.extractXmlValue(rootElement, "autoDeploymentsRegex")).thenReturn("regex");
+
+        // Act
+        listener.sendPluginConfigData(file);
+
+        // Assert
+        verify(pluginConfigApi).sendConnectionData(
+                eq("url"),
+                eq("secret"),
+                any(PipelineLogger.class),
+                anyString(),
+                eq(true),
+                eq("regex"),
+                eq(true),
+                eq("regex")
+        );
+    }
+
+}
```

Comment:
Should we have some more test cases here? Perhaps for when things aren't successful/edge cases?

Reply:
added a handful more tests and made a mockdatafunction :excite: